### PR TITLE
Adding Category Listing

### PR DESCRIPTION
**Pull request by MaharshM**

### CODE REVIEW

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -17,7 +18,9 @@ class CategoryController extends Controller
          */
         public function index()
         {
-            //
+            $categories =  Category::all();
```
Comment (abbasali) [inline app/Http/Controllers/CategoryController.php:21]: So any user can see all the categories? i.e. I can see other users' categories as well?

**Review by abbasali — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CategoryController.php (hunk 1 above)
Comment (MaharshM) [reply app/Http/Controllers/CategoryController.php]: done.

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -17,7 +18,10 @@ class CategoryController extends Controller
          */
         public function index()
         {
-            //
+
+            $categories =  DB::table('categories')->where('user_id', Auth::user()->id)->get();
```
Comment (abbasali) [inline app/Http/Controllers/CategoryController.php:22]: Use eloquent model to find categories. In fact, it is better to use relationship between user and category to get all categories of current user.

**Review by abbasali — CHANGES_REQUESTED** [review]

**Review by abbasali — CHANGES_REQUESTED** [review]

File: app/Http/Requests/CreateCategoryRequest.php [hunk 3]
```diff
@@ -23,10 +24,11 @@ public function authorize()
      */
     public function rules()
     {
+        $check_user = Auth::user()->id;
```
[the remark this answers is not in the document]
Comment (MaharshM) [reply app/Http/Requests/CreateCategoryRequest.php:27]: done.

**Review by abbasali — APPROVED** [review]

**Pull request merged**